Examples/Swift/ManageOfflineRegionsExample.swift

```diff
@@ -1,220 +1,234 @@
         }
 
+    }
 
-        func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-            let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
-            if MGLOfflineStorage.shared.packs?.count != nil {
-                let progress = Float(packArray[indexPath.row].progress.countOfResourcesCompleted)/Float(packArray[indexPath.row].progress.countOfResourcesExpected)
-                cell.textLabel!.text = "Region \(indexPath.row + 1): \n  Percent completion: \(String(format: "%.2f", progress * 100)) %"
-                print(Float((packArray[indexPath.row].progress.countOfResourcesCompleted)/(packArray[indexPath.row].progress.countOfResourcesExpected) * 100))
-
-                cell.textLabel?.font = UIFont.systemFont(ofSize: 15)
-                cell.textLabel?.numberOfLines = 0
-                cell.textLabel?.lineBreakMode = .byWordWrapping
-                if #available(iOS 13.0, *) {
-                    cell.textLabel?.textColor = .darkGray
-                } else {
-                    // Fallback on earlier versions
-                }
-                cell.textLabel?.textAlignment = .left
+    deinit {
+        // Remove offline pack observers.
+        print("Removing offline pack notification observers")
+        NotificationCenter.default.removeObserver(self)
 
-            } else {
+    }
 
-                cell.textLabel?.text = "No Offline Packs Saved"
-                cell.textLabel?.textColor = .darkGray
-                cell.textLabel?.textAlignment = .center
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
 
-            }
+        if let packs = MGLOfflineStorage.shared.packs {
+            return packs.count
+        } else {
+            return 1
+        }
+    }
 
-            return cell
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        var cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
+        cell = UITableViewCell(style: .subtitle, reuseIdentifier: "cell")
+        if let packs = MGLOfflineStorage.shared.packs {
+            let completed = Float(packs[indexPath.row].progress.countOfResourcesCompleted)
+                let expected = Float(packs[indexPath.row].progress.countOfResourcesExpected)
+                let progress = completed / expected
+            cell.textLabel?.text = "Region \(indexPath.row + 1): size: \(packs[indexPath.row].progress.countOfBytesCompleted)"
+            cell.detailTextLabel?.text = "Percent completion: \(String(format: "%.2f", progress * 100))%"
+            cell.textLabel?.textAlignment = .left
+
+        } else {
+            print("so...")
+            cell.textLabel?.text = "No Offline Packs Saved"
+            cell.textLabel?.textColor = .darkGray
+            cell.textLabel?.textAlignment = .center
```

Comment:
This would look nicer in a header. It would also be more efficient, since `cellForRowAt:indexPath` can potentially be called a lot of you have a lot of cells. If you already have at least one cell, then you know you've got packs, and so the rest of this if/else check is extra work.

`UITableViewDataSource` supports using titles to mark section headers. 

You can try something like:

```swift
func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
   /**
     If there are downloaded packs, return "Offline packs",
     otherwise return "No pack downloaded".
     */
}
```